api/java/org/openyolo/api/CredentialClient.java

```diff
@@ -137,30 +139,29 @@ public void retrieve(
      * be returned. Otherwise, {@link android.app.Activity#RESULT_CANCELED} will be returned.
      */
     @Nullable
-    public Intent getSaveIntent(Credential credentialToSave) {
-        List<ComponentName> saveProviders =
-                findProviders(SAVE_CREDENTIAL_ACTION);
+    public Intent getSaveIntent(final CredentialSaveRequest saveRequest) {
+        List<ComponentName> saveProviders = findProviders(SAVE_CREDENTIAL_ACTION);
 
         if (saveProviders.isEmpty()) {
             return null;
         }
 
-        byte[] encodedCredential = credentialToSave.toProtobuf().toByteArray();
+        final byte[] encodedSaveRequest = saveRequest.toProtocolBuffer().toByteArray();
```

Comment:
nit: final is superfluous and inconsistent with the definition of other local variables. I usually only specify final on a local if it is to be used inside a closure / anonymous inner class.